### PR TITLE
Test automation around YAML chomping

### LINKED ISSUE #8655: Automation: Test automation around YAML chomping
Based on the work done in https://github.com/rancher/dashboard/issues/5958, we should add test automation to confirm that the YAML chomping works as expected. @torchiaf can offer guidance on his work.

Unit test is likely the way to go.

This should be easy in most cases whereby we compare the output of the YAML in and then out but there may be edge cases to consider that add complexity.

### PR DESCRIPTION
**Pull request by torchiaf**
### Summary
Fixes #8655

### CODE REVIEW

File: shell/utils/__tests__/create-yaml.test.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+} from '@shell/utils/create-yaml';
+
+const key = 'example';
+const randomData = '\n      foo\n      bar\n';
+
+const scalarStyles = ['>', '|'];
+const chomping = ['+', '-', ''];
+const indentations = ['4', '2', ''];
+
+describe('fx: getBlockDescriptor', () => {
```
[the remark this answers is not in the document]
Comment (torchiaf) [reply shell/utils/__tests__/create-yaml.test.ts:13]: > why use `fx:`?

We use this format in the other unit tests, so the title is 'function:' + name of the function we are testing.

File: shell/utils/create-yaml.js [hunk 2]
```diff
@@ -464,15 +464,16 @@ export function dumpBlock(data, options = {}) {
 
   if (blockFields.length) {
     for (const key of blockFields) {
+      const desc = getBlockDescriptor(out, key);
+
       const scalarStyle = options[key]?.scalarStyle ?? '|';
       const chomping = options[key]?.chomping ?? '';
-
-      const desc = getBlockDescriptor(out, key);
+      const indentation = desc.indentation;
```
[the remark this answers is not in the document]
Comment (torchiaf) [reply shell/utils/create-yaml.js:471]: Agree, much better to use de-structured fields/
